lock-api/src/test/java/com/palantir/lock/client/LeasingTimelockClientTest.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.time.Duration;
+import java.util.Set;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.common.time.NanoTime;
+import com.palantir.lock.v2.IdentifiedTime;
+import com.palantir.lock.v2.LeasableLockResponse;
+import com.palantir.lock.v2.LeasableRefreshLockResponse;
+import com.palantir.lock.v2.LeasableStartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.Lease;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.LockRequest;
+import com.palantir.lock.v2.LockResponse;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionRequest;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.lock.v2.TimestampAndPartition;
+
+public class LeasingTimelockClientTest {
+    private TimelockRpcClient timelockService = mock(TimelockRpcClient.class);
+    private TimelockService timelockClient;
+
+    private static final UUID LEADER_ID = UUID.randomUUID();
+
+    private static final LockToken LOCK_TOKEN = LockToken.of(UUID.randomUUID());
+    private static final LockResponse LOCK_RESPONSE = LockResponse.successful(LOCK_TOKEN);
+
+    @Before
+    public void setUp() {
+        when(timelockService.getLeaderTime()).thenReturn(IdentifiedTime.of(UUID.randomUUID(), NanoTime.now()));
+        timelockClient = LeasingTimelockClient.create(timelockService);
+    }
+
+    @Test
+    public void delegatesLockRequest() {
+        LockRequest lockRequest = mock(LockRequest.class);
```

Comment:
don't use the `mock()` static method, instead add `@RunWith(MockitoJUnitRunner.class)` to the whole class and create your mocks as fields, like

```
@Mock private LockRequest lockRequest;
```

why? because Mockito will tell you if you're mocking things unnecessarily (i.e. that don't get called), which is usually a sign of bugs.